### DISCUSSION TITLE
Separately styling the base directory and current folder in prompt Path

### DISCUSSION THREAD

**Discussion by KyleKing** [JanDeDobbeleer/oh-my-posh#2587] [Troubleshoot] [answered]
I really liked how my `p10k` prompt would show the full path while highlighting the current directory. I found a way to do that in `oh-my-posh`:

![config](https://user-images.githubusercontent.com/3784339/181570484-0195645a-98b8-4058-b844-2fb220734fb8.png)

```json
{
    "blocks": [
        {
          "foreground": "p:white",
          "properties": {
            "command": "((pwd | grep -q \"$HOME/\") && (dirname \"$PWD\" | sed -E \"s-^$HOME($|(/.*))-~\\2-\")) || true",
            "shell": "sh"
          },
          "style": "plain",
          "template": "{{ .Output }}/",
          "type": "command"
        },
        {
          "foreground": "p:blue",
          "style": "plain",
          "template": "{{ .Folder }} ",
          "type": "text"
        }
    ]
}
```

Would love to see if there is an easier or more efficient way to go about it

**Comment by JanDeDobbeleer** [comment, marked as answer]
Interesting, let's create a style for this.

**Reply by JanDeDobbeleer** [reply]
with the change coming in #2601 you can do this:

```json
{
    "foreground": "p:white",
    "properties": {
        "style": "folder"
    },
    "style": "plain",
    "template": " {{ .Parent }}/<p:blue>{{ .Path }}</> ",
    "type": "path"
}
```

provided that

**Reply by KyleKing** [reply]
That's awesome, thanks!!

**Reply by KyleKing** [reply]
Actually, there are subtle differences when using the absolute parent path:

- My version had my home directory replaced with `~`
- And when used with `.Path`, `.Parent` needs to be `''` to avoid collision with `.Path`

<img width="237" alt="Screen Shot 2022-08-02 at 21 23 45" src="https://user-images.githubusercontent.com/3784339/182504122-599d00ab-6c55-4f28-8311-ac572e12aaed.png">

When I would have expected:

```sh
~ > ..
/Users > ..
/ > cd ~/Downloads
~/Downloads >
```

Would you be open to making changes to the `.Parent` so that maybe the template could be written like: `"{{if .Parent }}/{{else}}<p:blue>{{ .Path }}</> "`?

This was how I wrote `.Parent` in shell:

```sh
((pwd | grep -q "$HOME/") && (dirname "$PWD" | sed -E "s-^$HOME($|(/.*))-~\\2-")) || true
```

**Reply by KyleKing** [reply]
Thanks! I just got back from vacation and saw that you made the fixes (https://github.com/JanDeDobbeleer/oh-my-posh/commit/47265b1bc19714ca6ff6900d65657502fc6f714a)!

<img width="157" alt="Screen Shot 2022-08-23 at 19 18 05" src="https://user-images.githubusercontent.com/3784339/186282481-4cc8fd3c-aa1c-4aa4-bc3f-67d8388e8482.png">

Refactored Chezmoi config:

```json
{
    "blocks": [
        {
          "foreground": "p:white",
          "properties": {
            "style": "folder"
          },
          "style": "plain",
          "template": "{{ .Parent }}<p:blue>{{ .Path }}</> ",
          "type": "path"
        }
    ]
}
```
